dotcom-rendering/src/web/components/ShowMore.importable.tsx

```diff
@@ -0,0 +1,200 @@
+import { css } from '@emotion/react';
+import {
+	from,
+	neutral,
+	space,
+	visuallyHidden,
+} from '@guardian/source-foundations';
+import {
+	Button,
+	InlineError,
+	SvgCross,
+	SvgPlus,
+} from '@guardian/source-react-components';
+import { useState } from 'react';
+import type { DCRContainerPalette, FEFrontCard } from 'src/types/front';
+import { enhanceCards } from '../../model/enhanceCards';
+import { shouldPadWrappableRows } from '../lib/dynamicSlices';
+import { useApi } from '../lib/useApi';
+import { useOnce } from '../lib/useOnce';
+import { LI } from './Card/components/LI';
+import { UL } from './Card/components/UL';
+import { FrontCard } from './FrontCard';
+
+function decideButtonText({
+	isOpen,
+	loading,
+	containerTitle,
+}: {
+	isOpen: boolean;
+	loading: boolean;
+	containerTitle: string;
+}) {
+	if (isOpen && loading) return 'Loading';
+	if (isOpen) return `Less ${containerTitle}`;
+	return `More ${containerTitle}`;
+}
+
+type Props = {
+	containerTitle: string;
+	path: string;
+	containerId: string;
+	showAge: boolean;
+	containerPalette?: DCRContainerPalette;
+};
+
+export const ShowMore = ({
+	containerTitle,
+	path,
+	containerId,
+	showAge,
+	containerPalette,
+}: Props) => {
+	function findCardLinks() {
+		const containerNode = document.getElementById(containerId);
+		const containerLinksArray = Array.from(
+			containerNode?.querySelectorAll('a') ?? [],
+		);
+		return containerLinksArray.map(
+			(Element) => Element.attributes.getNamedItem('href')?.value,
+		);
+	}
+
+	const [cardLinks, setCardLinks] = useState<string[]>([]);
+
+	useOnce(() => {
+		const linksArray: string[] = findCardLinks().filter(
+			(item): item is string => !!item,
+		);
+		setCardLinks(linksArray);
+	}, [containerId]);
+
+	const [isOpen, setIsOpen] = useState(false);
+
+	function toggleOpen() {
+		setIsOpen(!isOpen);
+	}
+
+	/** We only pass an actual URL to SWR when 'showMore' is true.
+	 * Toggling 'showMore' will trigger a re-render
+	 *   @see https://swr.vercel.app/docs/conditional-fetching#conditional
+	 */
+	const url = isOpen
+		? `https://api.nextgen.guardianapps.co.uk/${path}/show-more/${containerId}.json?dcr=true`
```

Comment:
I don't think should be a blocker, but if anyone knows of an easy way to avoid hard-coding this URL then that could be good!

Reply:
I don't think its too problematic to have the URL here, but ideally we'd check if we were on a CODE front and switch to the CODE API domain instead of it always being prod

Reply:
I think there's an apiUrl in the config?

Reply:
that could be useful! do you know where it is @mxdvl? I haven't been able to find one. I was thinking of pulling the stage from `window.guardian.config.stage` and switching on that, but if the actual URL is available somewhere then that would be great 👍

Reply:
our current type definitions seem to assume that `window.guardian.config.page.ajaxUrl` will always return a string, in which case we wouldn't need to provide a default hard-coded value (here I'm storing it in state because afaik `window` access within islands needs to happen within `useOnce`). And we also don't need to check whether it does return a string at runtime before we use it.

I don't know too much about the `window.guardian` setup though -- do those seem like reasonable assumptions or should we be building in more safeguards when using it? cc. @AshCorr & @mxdvl 

this is the sort of thing I've got at the moment, and the type checker seems happy with it:

```js
	const [baseUrl, setBaseUrl] = useState(
		'https://api.nextgen.guardianapps.co.uk',
	);

	useOnce(() => {
                 // snip
		setBaseUrl(window.guardian.config.page.ajaxUrl);
	}, []);

	// snip
	const url = isOpen
		? `${baseUrl}/${path}/show-more/${containerId}.json?dcr=true`
		: undefined;
```

Reply:
This is coming through as a prop on the object, so you also use it as a prop!

Reply:
oh great, that's a much better solution! thanks!